### PR TITLE
[FLINK-32912][release] Build 1.18 docs in GitHub Action

### PR DESCRIPTION
**Pull request by JingGe**

### CODE REVIEW

**Review by snuyanzin — APPROVED** [review]
LGTM

**Review by PatrickRen — COMMENTED** [review]
LGTM, but I think this should be merged when we finalize the release after the RC and voting procedure.

**Comment by JingGe** [comment]
@flinkbot run azure

**Comment by JingGe** [comment]
@flinkbot run azure

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -42,8 +42,8 @@ jobs:
           echo "flink_branch=${currentBranch}" >> ${GITHUB_ENV}
 
           if [ "${currentBranch}" = "master" ]; then
-            echo "flink_alias=release-1.18" >> ${GITHUB_ENV}
-          elif [ "${currentBranch}" = "release-1.17" ]; then
```
Comment (PatrickRen) [inline .github/workflows/docs.yml:46]: We need to keep using `release-1.17` here as the latest stable version, and change it to 1.18 after we finally release the version.

**Review by PatrickRen — CHANGES_REQUESTED** [review]
@JingGe I checked the history in 1.16. We need to keep release-1.17 as the latest stable version for now.

File: .github/workflows/docs.yml (hunk 1 above)
Comment (leonardBang) [reply .github/workflows/docs.yml:46]: +1

Comment (JingGe) [reply .github/workflows/docs.yml:46]: got it, thanks!

Comment (JingGe) [reply .github/workflows/docs.yml:46]: I have addressed your suggestion. Please check it again, thanks!

**Review by PatrickRen — APPROVED** [review]
LGTM. Could you rebase the latest master and wait for the CI result (at least the compile stage)?

**Pull request merged**